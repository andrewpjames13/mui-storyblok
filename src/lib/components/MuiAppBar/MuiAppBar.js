/* eslint-disable max-len */
// import React, { createElement, lazy, Suspense } from 'react';
import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';

import Storyblok from '../../utils/Storyblok';
import MuiButtonRedirect from '../MuiButtonRedirect/MuiButtonRedirect';
import MuiButtonHref from '../MuiButtonHref/MuiButtonHref';
import MuiIconButtonRedirect from '../MuiIconButtonRedirect/MuiIconButtonRedirect';
import MuiIconButtonHref from '../MuiIconButtonHref/MuiIconButtonHref';
import MuiMenu from '../MuiMenu/MuiMenu';
import MuiButtonDrawer from '../MuiButtonDrawer/MuiButtonDrawer';
import MuiIconButtonDrawer from '../MuiIconButtonDrawer/MuiIconButtonDrawer';
import MuiButtonSnackbar from '../MuiButtonSnackbar/MuiButtonSnackbar';

// Styleguidist uses react-docgen and they do not support ImportExpression yet and was unable to generat docs for props
// leaving code her as a reminder that it needs to be done once support comes out for it
// const MuiButtonRedirect = lazy(() => import('../MuiButtonRedirect/MuiButtonRedirect'));
// const MuiButtonHref = lazy(() => import('../MuiButtonHref/MuiButtonHref'));
// const MuiIconButtonRedirect = lazy(() => import('../MuiIconButtonRedirect/MuiIconButtonRedirect'));
// const MuiIconButtonHref = lazy(() => import('../MuiIconButtonHref/MuiIconButtonHref'));
// const MuiMenu = lazy(() => import('../MuiMenu/MuiMenu'));

// <Suspense fallback={<div />}>
//   {content.map((item, index) => createElement(
//     components[item.component],
//     Object.assign(item, { key: index }),
//   ))}
// </Suspense>;

const MuiAppBar = ({
  rootClass,
  color,
  position,
  content,
  variant,
  disableGutters,
}) => {
  const components = {
    MuiButtonRedirect,
    MuiButtonHref,
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiMenu,
    MuiButtonDrawer,
    MuiIconButtonDrawer,
    MuiButtonSnackbar,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <AppBar
      className={styles.root}
      color={color}
      position={position}
    >
      <Toolbar
        variant={variant}
        disableGutters={disableGutters}
      >
        {content.map((item, index) => createElement(
          components[item.component],
          Object.assign(item, { key: index }),
        ))}
      </Toolbar>
    </AppBar>
  );
};

export default MuiAppBar;

MuiAppBar.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component.
  */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: 'default' | 'inherit' | 'primary' | 'secondary' | 'transparent'
   * AppBar: The color of the component. It supports those theme colors that make sense for this component.
   * */
  color: PropTypes.string,
  /**
   * 'absolute'| 'fixed'| 'relative'| 'static' | 'sticky'
   * AppBar: The positioning type. The behavior of the different options is described in the MDN web docs.
   * Note: sticky is not universally supported and will fall back to static when unavailable.
   * */
  position: PropTypes.string,
  /**
   * mui prop: 'regular'| 'dense'
   * ToolBar: The variant to use.
   *  */
  variant: PropTypes.string,
  /**
   * mui prop: true | false
   * Toolbar: If true, disables gutter padding.
   * */
  disableGutters: PropTypes.bool,

  /**
   * Content passed to MuiGrid to render
   * components: MuiTypography,
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiButtonRedirect,
    MuiButtonHref,
    MuiMenu,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiAppBar.defaultProps = {
  color: 'primary',
  position: 'fixed',
  variant: 'regular',
  disableGutters: false,
  rootClass: [],
};
