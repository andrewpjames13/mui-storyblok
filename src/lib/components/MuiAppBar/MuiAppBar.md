MuiAppBar example:

```js
const content = [
  {
    anchor: 'left',
    component: 'MuiIconButtonDrawer',
    variant: 'temporary',
    content: [{
      component: 'MuiList',
      content: [{
        component: 'MuiListItem',
        listItemSecondaryAction: [{
          component: 'MuiListItemSecondaryAction',
          content: [{
            component: 'MuiIconButtonRedirect',
            redirectRoute: '/',
            iconButton: [{
              component: 'MuiIconButton',
              icon: [{
                component: 'MuiIcon',
                iconName: 'home',
              }],
            }],
          }],
        }],
        listItemText: [{
          component: 'MuiListItemText',
          primary: 'Home',
        }],
      }],
    }],
    icon: [{
      component: 'MuiIconButton',
      size: 'medium',
      icon: [{
        color: 'secondary',
        component: 'MuiIcon',
        iconName: 'menu',
      }],
    }],
  },
  {
    btnText: 'MuiMenu',
    component: 'MuiMenu',
    content: [{
      buttonText: 'Menu Item',
      component: 'MuiMenuItem',
      redirectRoute: '/',
    }],
  },
  {
    component: 'MuiButtonRedirect',
    redirectRoute: '',
    button: [{
      component: 'MuiButton',
      buttonText: 'here',
    }],
  },
];

  <MuiAppBar
    position="relative"
    content={content}
    disableGutters={false}
    variant="regular"
  />
```