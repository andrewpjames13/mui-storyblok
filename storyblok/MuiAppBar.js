module.exports = {
  "name": "MuiAppBar",
  "display_name": null,
  "updated_at": "2020-06-12T15:30:19.658Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "use_uuid": true,
      "description": "Override or extend the styles applied to the component.",
      "options": [],
      "pos": 0
    },
    "height": {
      "type": "text",
      "description": "Height of the App Bar",
      "default_value": "60px",
      "pos": 1
    },
    "color": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "default",
          "name": "default"
        },
        {
          "value": "inherit",
          "name": "inherit"
        },
        {
          "value": "primary",
          "name": "primary"
        },
        {
          "value": "secondary",
          "name": "secondary"
        },
        {
          "value": "transparent",
          "name": "transparent"
        }
      ],
      "required": true,
      "default_value": "primary",
      "description": "AppBar: The color of the component. It supports those theme colors that make sense for this component.",
      "pos": 2
    },
    "position": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "absolute",
          "name": "absolute"
        },
        {
          "value": "fixed",
          "name": "fixed"
        },
        {
          "value": "relative",
          "name": "relative"
        },
        {
          "value": "static",
          "name": "static"
        },
        {
          "value": "sticky",
          "name": "sticky"
        }
      ],
      "default_value": "fixed",
      "required": true,
      "description": "AppBar: The positioning type. The behavior of the different options is described in the MDN web docs.    * Note: sticky is not universally supported and will fall back to static when unavailable.",
      "pos": 3
    },
    "variant": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "regular",
          "name": "regular"
        },
        {
          "value": "dense",
          "name": "dense"
        }
      ],
      "required": true,
      "default_value": "regular",
      "description": "ToolBar: The variant to use.",
      "pos": 4
    },
    "disableGutters": {
      "type": "boolean",
      "description": "Toolbar: If true, disables gutter padding.",
      "pos": 5
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiAppBarGrid"
      ],
      "required": true,
      "pos": 6
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiAppBar",
  "component_group_uuid": null
}