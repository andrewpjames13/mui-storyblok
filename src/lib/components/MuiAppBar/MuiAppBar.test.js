import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiAppBar from './MuiAppBar';

function setup() {
  const props = {
    content: [{
      component: 'GridItem',
      content: [{
        component: 'MuiButtonHref',
        button: [{
          component: 'MuiButton',
          buttonText: 'cool',
        }],
        href: 'cool.com',
      }],
    }],
  };
  const comp = shallow(<MuiAppBar {...props} />);
  return { comp, props };
}

describe('<MuiAppBar />', () => {
  it('renders MuiAppBar', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiAppBar {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
