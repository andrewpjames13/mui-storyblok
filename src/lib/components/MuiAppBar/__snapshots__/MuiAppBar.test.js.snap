// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiAppBar /> snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-root-3 mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-4 MuiButton-textSecondary MuiButton-fullWidth"
      data-testid="muiButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        here
      </span>
    </button>
  </div>
</header>
`;
